docs/DevGuide/Travis.md

```diff
@@ -74,6 +74,30 @@ performed in the CHECK_FILES build.
 using a Linux OS, and the `AppleClang` compiler for `OS X`.
 * The `gcc Debug` build will fail if there are `doxygen` warnings.
 
+## How to perform the checks locally
+
+Before pushing to GitHub and waiting for Travis to perform the checks it is
+useful to perform at least the following tests locally:
+- **Unit tests:** Perform a `make test-executables` and then execute `ctest` to
+  run all unit tests. As for `make` you can append a `-jN` flag to `ctest` to
+  run in parallel on `N` cores. To run only a subset of the tests you can use
+  one of the other keywords that the tests are labeled with, such as `ctest -L
+  datastructures`. To run only particular tests you can also execute `ctest -R
+  TEST_NAME` instead, where `TEST_NAME` is a regular expression matching the
+  test identifiers such as `Unit.DataStructures.Mesh`. Pass the flag
+  `--output-on-failure` to get output from failed tests. Consult `ctest -h` for
+  further options.
+- **clang-tidy:** Run `make clang-tidy FILE=SOURCE_FILE` where `SOURCE_FILE` is
```

Comment:
Can we also state here that clang-tidy must be run in a clang build? Perhaps open the paragraph with `In a clang build, run [...]`?

Reply:
Can you make clang-tidy automatically run whenever you compile? Is there a reason not to do that?

Reply:
I mean, run clang-tidy on the files that changed only, to save time...

Reply:
It's still quite slow; I would not want clang-tidy to run without having asked it to.

Reply:
Exactly what @fmahebert said. It would result in compilation taking 2 or 3 hours instead of 10 minutes...

Reply:
I'm fine with not running this automatically because it's slow. I'll just run clang-tidy-hash when I want to check this. 

So I understand: @nilsdeppe, you are talking about 2-3 hours to run clang-tidy on everything, while clang-tidy-hash (so only changed files get checked) takes maybe 5-6 minutes, which is still too long to want to run it every time?

Reply:
Well, say I compile my code every 3 minutes on average because I'm developing, and currently it takes 15 seconds to do an incremental build, my total turnaround time is 3:15 min. If you want to add 5 minutes to that I will never get anything done, nor will anyone else. I think this would be a completely unacceptable regression and nobody will agree to it. Running clang-tidy-hash and iwyu-hash before submitting a PR is a reasonable thing to ask people to do in comparison.